packages/clarity-decode/src/data.ts

```diff
@@ -45,6 +45,13 @@ export function decode(tokens: Data.Token[]): DataEvent {
                 activityTime: tokens[11] as number
             }
             return { time, event, data: baselineData };
+        case Data.Event.Variable:
+            let v = 2; // Start from 3rd index since first two are used for time & event
+            let variables: Data.VariableData = {};
+            while (v < tokens.length) {
```

Comment:
>tokens.length [](start = 23, length = 13)

should this be tokens.length + 1

Reply:
May be I am missing something here... but what we have looks right. If I add +1 then we will attempt to read tokens[tokens.length] which will be undefined.

Reply:
to me too.

---
In reply to: [443007284](https://github.com/microsoft/clarity/pull/24#discussion_r443007284) [](ancestors = 443007284)

Reply:
sorry i meant -1 since we are incrementing twice inside the loop 

---
In reply to: [444364988](https://github.com/microsoft/clarity/pull/24#discussion_r444364988) [](ancestors = 444364988,443007284)